packages/@sanity/default-layout/src/navbar/search/SearchDialog.tsx

```diff
@@ -0,0 +1,199 @@
+
+  const hasValidTerms = hasSearchableTerms(terms)
+
+  /**
+   * Measure top-most visible search result index
+   */
+  const {lastSearchIndex, resetLastSearchIndex, setLastSearchIndex} = useMeasureSearchResultsIndex(
+    childContainerElement
+  )
+
+  /**
+   * Reset last search index when new results are loaded, or visiting recent searches
+   * @todo Revise if/when we introduce pagination
+   */
+  useEffect(() => {
+    if (!isMountedRef?.current) {
+      isMountedRef.current = true
+      return
+    }
+
+    if (!hasValidTerms || result.loaded) {
+      resetLastSearchIndex()
+    }
+  }, [hasValidTerms, resetLastSearchIndex, result.loaded])
+
+  /**
+   * Store top-most search result scroll index on close
+   */
+  const handleClose = useCallback(() => {
+    setLastSearchIndex()
+    onClose()
+  }, [onClose, setLastSearchIndex])
+
+  /**
+   * Bind hotkeys to open / close actions
+   */
+  useSearchHotkeys({onClose: handleClose, onOpen, open})
+
+  if (!open) {
+    return null
+  }
+
+  return (
+    <CommandListProvider
+      ariaChildrenLabel={hasValidTerms ? 'Search results' : 'Recent searches'}
+      ariaHeaderLabel="Search"
+      autoFocus
+      childContainerElement={childContainerElement}
+      childCount={hasValidTerms ? result.hits.length : recentSearches.length}
+      containerElement={containerElement}
+      headerInputElement={headerInputElement}
+      id={ID}
+      initialSelectedIndex={hasValidTerms ? lastSearchIndex : 0}
+      pointerOverlayElement={pointerOverlayElement}
+      virtualList={hasValidTerms}
+    >
+      <Portal>
+        <FocusLock autoFocus={false} returnFocus>
+          <SearchDialogWrapper ref={setContainerRef} scheme="light" tone="default">
+            <Flex direction="column" height="fill">
+              <SearchHeader onClose={handleClose} setHeaderInputRef={setHeaderInputRef} />
+              <SearchContent flex={1}>
+                {hasValidTerms ? (
+                  <SearchResults
+                    onClose={handleClose}
+                    setChildContainerRef={setChildContainerRef}
+                    setPointerOverlayRef={setPointerOverlayRef}
+                  />
+                ) : (
+                  <RecentSearches
+                    setChildContainerRef={setChildContainerRef}
+                    setPointerOverlayRef={setPointerOverlayRef}
+                  />
+                )}
+              </SearchContent>
+
+              <SearchDialogFilters />
+            </Flex>
+          </SearchDialogWrapper>
+        </FocusLock>
+      </Portal>
+    </CommandListProvider>
+  )
+}
+
+function SearchDialogFilters() {
+  const {
+    dispatch,
+    state: {filtersVisible},
+  } = useSearchState()
+
+  const dialogRef = useRef<HTMLDivElement>(null)
+
+  // Always hide filters on mount
+  useEffect(() => {
+    dispatch({type: 'FILTERS_HIDE'})
+  }, [dispatch])
+
+  // Force dialogs to be 100% height
+  // TODO: there has to be a better way to do this
```

Comment:
If we dont find a better way, do we need the todo?

Reply:
Also, the dep filtersVisible is not used. I guess its driving this effect to trigger, but then we perhaps only need to set height when filtersVisible is true?

Reply:
Good point! Cleaned this to be slightly less offensive, also removed useEffect dependency as well as the redundant dispatch